### PR TITLE
[WIP] Implement code which integrate openshift CI and Aws s3

### PR DESCRIPTION
**Pull request by flacatus**
The current PR add new enhance to che-test-harness like:

- Transform json metadata with test results into prometheus file to use later to push to the Prometheus Push Gateway
- Use aws client go to send  the prometheus file to S3 buckets
- Some improvements in code base
- Measure times of three workspaces (simple, nodejs and java maven)

Signed-off-by: flacatus <flacatus@redhat.com>

### CODE REVIEW

File: cmd/che/che_harness_test.go [hunk 1]
```diff
@@ -57,6 +65,27 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 }, func(data []byte) {})
 
 var _ = ginkgo.SynchronizedAfterSuite(func() {
```
Comment (Katka92) [inline cmd/che/che_harness_test.go:67]: What about using some meaningful name instaed of `_`?

File: cmd/che/che_harness_test.go [hunk 2]
```diff
@@ -77,27 +106,42 @@ var _ = ginkgo.SynchronizedAfterSuite(func() {
...
-	Logger.Info("Running Eclipse Che e2e tests...")
 	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "Eclipse Che Test Harness", r)
 
-	err := metadata.Instance.WriteToJSON(filepath.Join(config.TestHarnessConfig.Artifacts, addonMetadataName))
+}
+
+// Register All flags used by test harness
+func registerCheFlags(flags *flag.FlagSet) {
+	flags.StringVar(&CfgInstance.artifactsDir, "artifacts-dir", "/tmp/artifacts", "If is specified test harness will save all reports in the given directory, if not will save artifacts in the current directory. Default dir is /tmp/artifacts")
+	flags.StringVar(&CfgInstance.metricsFiles, "metrics-files", "", "If it is set che test harness start to send the data to AWS S3 . You should have valid secrets in the files")
```
Comment (Katka92) [inline cmd/che/che_harness_test.go]: Redundandt space after `AWS S3`.

File: cmd/che/che_harness_test.go [hunk 3]
```diff
@@ -77,27 +106,42 @@ var _ = ginkgo.SynchronizedAfterSuite(func() {
...
+}
+
+// Generate a Prometheus file from json metadata and connect with aws s3 and put the data into buckets
+func sendDataToS3(prometheusFilename string) error {
+	if err := common.LoadConfigs(CfgInstance.metricsFiles); err != nil {
+		return fmt.Errorf("error loading initial state: %v", err)
+	}
+
+	if err := uploadFileToMetricsBucket(filepath.Join(CfgInstance.artifactsDir, prometheusFilename)); err != nil {
+		return fmt.Errorf("error while uploading prometheus prometheus: %v", err)
```
Comment (Katka92) [inline cmd/che/che_harness_test.go]: Typo - maybe you mean `prometheus metrics`?

File: pkg/controller/workspaces/workspaces.go [hunk 4]
```diff
@@ -150,18 +152,34 @@ func (w *WorkspacesController) StartWorkspace(token string, cheURL string, works
 	return err
 }
 
+// StartWorkspace delete a from a given workspace_id
```
Comment (Katka92) [inline pkg/controller/workspaces/workspaces.go]: Wrong description.

File: pkg/suites/che/workspaces_tests.go [hunk 5]
```diff
@@ -32,13 +38,13 @@ var _ =  KubeDescribe( "[Workspaces]" , func() {
 		_ = ctrl.RunWorkspace(file, workspaceStack)
 	})
 
-	ginkgo.It("Start Python 3.7 Workspace", func() {
-		workspaceStack := "python"
+	ginkgo.It("Start NodeJS Workspace", func() {
+		workspaceStack := "nodejs"
 		httpClient, err := client.NewHttpClient()
```
Comment (Katka92) [inline pkg/suites/che/workspaces_tests.go:43]: In Java Maven test, there is a sleep command. Can you please explain, why it is missing for the NodeJS test?

**Review by Katka92 — APPROVED** [review]

File: pkg/suites/che/workspaces_tests.go (hunk 5 above)
Comment (flacatus) [reply pkg/suites/che/workspaces_tests.go:43]: Oh, Good catch! I was doing some tests locally and I forgot to remove this sleep command.

File: cmd/che/che_harness_test.go (hunk 1 above)
Comment (flacatus) [reply cmd/che/che_harness_test.go:67]: Is an unused variable, so better to have `_`

File: pkg/controller/workspaces/workspaces.go (hunk 4 above)
Comment (flacatus) [reply pkg/controller/workspaces/workspaces.go]: Fixed

**Pull request merged**